api/v1alpha1/baseboardmanagement_types.go

```diff
@@ -21,54 +21,74 @@ import (
+	Pxe   BootDevice = "pxe"
+	Disk  BootDevice = "disk"
+	Bios  BootDevice = "bios"
+	Cdrom BootDevice = "cdrom"
+	Safe  BootDevice = "safe"
+)
 
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
 // BaseboardManagementSpec defines the desired state of BaseboardManagement
 type BaseboardManagementSpec struct {
-	// Host is the host IP address of the BaseboardManagement.
-	// +kubebuilder:validation:MinLength=1
-	Host string `json:"host"`
 
-	// AuthSecretRef is the SecretReference that contains authentication information of the BaseboardManagement.
-	// The Secret must contain username and password keys.
-	AuthSecretRef corev1.SecretReference `json:"authSecretRef"`
+	// Connection represents the BaseboardManagement connectivity information.
+	Connection Connection `json:"connection"`
+
+	// Power is the desired power state of the BaseboardManagement.
+	// +kubebuilder:validation:Enum=On;Off
+	Power PowerState `json:"power"`
 
 	// Vendor is the vendor name of the BaseboardManagement.
 	// +kubebuilder:validation:MinLength=1
 	Vendor string `json:"vendor"`
-
-	// Power is the desired power state of the BaseboardManagement.
-	// +kubebuilder:validation:MinLength=1
-	Power string `json:"power"`
-
-	// Boot is the desired boot device for the BaseboardManagement.
-	Boot Boot `json:"boot"`
 }
 
-type Boot struct {
-	// BootDevice is the desired boot for the BaseboardManagement.
+type Connection struct {
+	// Host is the host IP address or hostname of the BaseboardManagement.
 	// +kubebuilder:validation:MinLength=1
-	BootDevice string `json:"bootDevice"`
+	Host string `json:"host"`
 
-	// Persistent if True, boot device is permanently set for the BaseboardManagement.
-	// If False, boot device is set as one time boot.
-	// +kubebuilder:default=false
-	Persistent bool `json:"persistent,omitempty"`
+	// AuthSecretRef is the SecretReference that contains authentication information of the BaseboardManagement.
+	// The Secret must contain username and password keys.
+	AuthSecretRef corev1.SecretReference `json:"authSecretRef"`
 
-	// EfiBoot specifies to EFI boot for the BaseboardManagement.
-	// +kubebuilder:default=false
-	EfiBoot bool `json:"efiBoot,omitempty"`
+	// InsecureTLS specifies trusted TLS connections.
+	InsecureTLS bool `json:"insecureTLS"`
 }
 
 // BaseboardManagementStatus defines the observed state of BaseboardManagement
 type BaseboardManagementStatus struct {
-	//+optional
-	Power BaseboardManagementState `json:"powerState,omitempty"`
-	//+optional
-	Boot BaseboardManagementState `json:"bootState,omitempty"`
+	// Power is the current power state of the BaseboardManagement.
+	// +kubebuilder:validation:Enum=On;Off
+	// +optional
+	Power PowerState `json:"powerState,omitempty"`
+
+	// BootDevice is the current first BootDevice of the BaseboardManagement.
+	// +optional
+	// +kubebuilder:validation:Enum=Pxe;Disk;Bios;Cdrom;Safe
+	BootDevice BootDevice `json:"bootState,omitempty"`
```

Comment:
If we do this and later implement the ability to retrieve the full boot order we're going to end up with an API change because: (1) we'll either change the semantics of this field and probably the type; or (2) we'll introduce a new field to represent the full boot order and need to support this or make a breaking change. 

Neither of those options are pleasant so it begs a question: do we reasonably think this will be useful in the short term? If not, I'd advocate removing it to alleviate any compatibility issues down the road.

Reply:
Makes sense. My idea here was just to note the status, but its true we would eventually want to change this. I can take it out for now.